src/main/java/com/ibm/domain/scanning/ScanAggregate.java

```diff
@@ -35,45 +40,70 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public final class ScanAggregate extends AggregateRoot<ScanId> {
-    private static final Logger LOGGER = LoggerFactory.getLogger(ScanAggregate.class);
+    // private static final Logger LOGGER = LoggerFactory.getLogger(ScanAggregate.class);
 
-    @Nonnull private final ScanRequest scanRequest;
+    @Nonnull private ResolvedScanRequest request;
     @Nullable private Commit commit;
```

Comment:
```suggestion
    @Nullable private GitUrl giturl;
    @Nullable private Purl purl;
    @Nullable private Commit commit;
```

Reply:
Kept request as is and added new mutable state attributes (@Nullable purl, @Nullable gitUrl, @Nonnull revision) to ScanAggregate.